src/scripts/sync_synonym.py

```diff
@@ -0,0 +1,166 @@
+    """Filter out all rows are not in common, given a `join_on`."""
+    merge_df = base_df.merge(intersecting_df, on=join_on, how='left', indicator=True)
+    filtered_df = merge_df[merge_df['_merge'] == 'left_only'].drop(columns=['_merge']).rename(
+        columns=lambda x: x[:-2] if x.endswith('_x') else x)[list(base_df.columns)]
+    return filtered_df
+
+
+# TODO
+def sync_synonyms(
+    ontology_db_path: Union[Path, str], mondo_synonyms_path: Union[Path, str],
+    mondo_mappings_path: Union[Path, str], onto_config_path: Union[Path, str], outpath_added: Union[Path, str],
+    outpath_confirmed: Union[Path, str], outpath_deleted: Union[Path, str], outpath_updated: Union[Path, str]
+):
+    """Create outputs for syncing synonyms between Mondo and its sources."""
+    # Iterate over mappings
+    mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#')
+    mappings_df = mappings_df[mappings_df['predicate_id'] == 'skos:exactMatch']
+    # todo: refactor filtering by prefixes
+    #  check if this pattern has been used elsewhere; I think it has. and abstract into func for these 3 lines; 'filter prefixes'
+    #  - OR use this?: get_all_owned_terms()
+    #    - it seems long and messy compared to these 3 lines, but is more comprehensive. I could make get_all_owned_terms() a simple 3 line func and then make a separate func for all edge cases?
+    # - filter by source
+    owned_prefix_map: PREFIX_MAP = get_owned_prefix_map(onto_config_path)
+    mappings_df['object_prefix'] = mappings_df['object_id'].apply(lambda x: x.split(':')[0])
+    mappings_df = mappings_df[mappings_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mappings_df['object_prefix']
+
+    # Query synonyms
+    # - Mondo
+    mondo_df = pd.read_csv(mondo_synonyms_path, sep='\t').rename(columns={'?dbXref': 'source_id'})
+    mondo_df = mondo_df.rename(columns=lambda x: x.lstrip('?'))
+    mondo_df['object_prefix'] = mondo_df['source_id'].apply(lambda x: x.split(':')[0])
+    mondo_df = mondo_df[mondo_df['object_prefix'].isin(owned_prefix_map.keys())]
+    del mondo_df['object_prefix']
+    mondo_df['unprefixed_source_id'] = mondo_df['source_id'].apply(lambda x: x.split(':')[1])
+    mondo_df = mondo_df[mondo_df['unprefixed_source_id'].str.isnumeric()]
+    del mondo_df['unprefixed_source_id']
+
+    # - Source
+    source_db: SqlImplementation = get_adapter(ontology_db_path)
+    source_df: pd.DataFrame = _query_synonyms(mappings_df['object_id'].tolist(), source_db)
+    #   - add mapped mondo ids
+    source_mondo_lookup = mappings_df.set_index('object_id')['subject_id'].to_dict()
+    source_df['mondo_id'] = source_df['curie'].map(source_mondo_lookup)
+    source_df = source_df.rename(columns={'curie': 'source_id'})[['mondo_id', 'source_id', 'synonym_scope', 'synonym']]
+    #   - obsoletes
+    # TODO: fully utilize obsoletes. see pr thread
+    source_obsoletes: Set[CURIE] = set([x for x in source_db.obsoletes()])
```

Comment:
## Obsoletes: `is_deprecated` column
For some or all of the 'case' templates, it may be useful to include this column.

Not that this is going to be what I do in the interim: I'll keep the obsoletes in all of the templates, pending our decisions, but I'll add this column for now.

Reply:
Since there are Mondo terms that are mapped to obsolete terms based on the contents of the `mondo.sssom.tsv` file, I would first check if the source term is obsolete and if it is ignore it from any further checks and save all of these source obsolete terms so they can be saved to a file for review.

Reply:
Implemented option (i.b.), purely exclude obsoletes from all cases:

```py
    source_obsoletes: Set[CURIE] = set([x for x in source_db.obsoletes()])
    mappings_df = mappings_df[~mappings_df['object_id'].isin(source_obsoletes)]
```

Reply:
- [ ] Review new outputs and ensure obsoletes were correctly filtered out

Reply:
@twhetzel I did a few checks and this seems to have been done correctly. Marking resolved.